packages/component-base/src/data-provider-controller/helpers.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Retreives information for the given flattened index, including:
+ * - the corresponding cache
+ * - the associated item (if loaded)
+ * - the corresponding index in the cache's items array.
+ * - the page containing the index.
+ * - the cache level
+ *
+ * @param {import('./cache.js').Cache} cache
+ * @param {number} flatIndex
+ * @return {{ cache: Cache, item: object | undefined, index: number, page: number, level: number }}
+ */
+export function getFlatIndexInfo(cache, flatIndex, level = 0) {
```

Comment:
**question:** Should these functions be separate helpers or methods of the `Cache` class?